lib/twap/events/self_interval_tick.js

```diff
@@ -71,11 +79,11 @@ const onSelfIntervalTick = async (instance = {}) => {
 
   let orderPrice
 
-  if (!/MARKET/.test(orderType)) {
+  if (!isMarketOrder) {
     if (hasTradeTarget(args)) {
       orderPrice = getTradePrice(state)
     } else if (hasOBTarget(args)) {
-      orderPrice = getOBPrice(state)
+      orderPrice = getOBPrice(_get(instance, 'state', {}))
```

Comment:
why using lodash get here? wouldn't we have a serious problem anyway if state is not set at this point?

Reply:
Since we updated the state with `isBookUpdated` to be false, the local `state` which we defined before updating the state still has `isBookUpdated: true`. So, reading the state here from the `instance` reference. Lodash get was just a preference over `instance.state`. `getOBPrice` will return null if the book isn't updated resulting in orderPrice to be null and scheduleTick function will be called.